typed/remove_test.go

```diff
@@ -0,0 +1,275 @@
+      type:
+        scalar: boolean
+    - name: nv
+      type:
+        scalar: numeric
+- name: myValue
+  map:
+    elementType:
+      scalar: string
+`
+
+var removeCases = []removeTestCase{{
+	name:         "simple pair",
+	rootTypeName: "stringPair",
+	schema:       typed.YAMLObject(simplePairSchema),
+	quadruplets: []removeQuadruplet{{
+		`{"key":"foo"}`,
+		_NS(_P("key")),
+		``,
+		`{"key":"foo"}`,
+	}, {
+		`{"key":"foo"}`,
+		_NS(),
+		`{"key":"foo"}`,
+		``,
+	}, {
+		`{"key":"foo","value":true}`,
+		_NS(_P("key")),
+		`{"value":true}`,
+		`{"key":"foo"}`,
+	}, {
+		`{"key":"foo","value":{"a": "b"}}`,
+		_NS(_P("value")),
+		`{"key":"foo"}`,
+		`{"value":{"a": "b"}}`,
+	}},
+}, {
+	name:         "struct grab bag",
+	rootTypeName: "myStruct",
+	schema:       typed.YAMLObject(structGrabBagSchema),
+	quadruplets: []removeQuadruplet{{
+		`{"setBool":[false]}`,
+		_NS(_P("setBool", _V(false))),
+		// is this the right remove output?
+		`{"setBool":null}`,
+		`{"setBool":[false]}`,
+	}, {
+		`{"setBool":[false]}`,
+		_NS(_P("setBool", _V(true))),
+		`{"setBool":[false]}`,
+		`{"setBool":null}`,
+	}, {
+		`{"setBool":[true,false]}`,
+		_NS(_P("setBool", _V(true))),
+		`{"setBool":[false]}`,
+		`{"setBool":[true]}`,
+	}, {
+		`{"setBool":[true,false]}`,
+		_NS(_P("setBool")),
+		``,
+		`{"setBool":[true,false]}`,
+	}, {
+		`{"setNumeric":[1,2,3,4.5]}`,
+		_NS(_P("setNumeric", _V(1)), _P("setNumeric", _V(4.5))),
+		`{"setNumeric":[2,3]}`,
+		`{"setNumeric":[1,4.5]}`,
+	}, {
+		`{"setStr":["a","b","c"]}`,
+		_NS(_P("setStr", _V("a"))),
+		`{"setStr":["b","c"]}`,
+		`{"setStr":["a"]}`,
+	}},
+}, {
+	name:         "associative list",
+	rootTypeName: "myRoot",
+	schema:       typed.YAMLObject(associativeListSchema),
+	quadruplets: []removeQuadruplet{{
+		`{"list":[{"key":"a","id":1},{"key":"a","id":2},{"key":"b","id":1}]}`,
+		_NS(_P("list", _KBF("key", "a", "id", 1))),
+		`{"list":[{"key":"a","id":2},{"key":"b","id":1}]}`,
+		`{"list":[{"key":"a","id":1}]}`,
+	}, {
+		`{"atomicList":["a", "a", "a"]}`,
+		_NS(_P("atomicList")),
+		``,
+		`{"atomicList":["a", "a", "a"]}`,
+	}},
+}}
+
+func (tt removeTestCase) test(t *testing.T) {
+	parser, err := typed.NewParser(tt.schema)
+	if err != nil {
+		t.Fatalf("failed to create schema: %v", err)
+	}
+
+	for i, quadruplet := range tt.quadruplets {
+		quadruplet := quadruplet
+		t.Run(fmt.Sprintf("%v-valid-%v", tt.name, i), func(t *testing.T) {
+			t.Parallel()
+			pt := parser.Type(tt.rootTypeName)
+
+			tv, err := pt.FromYAML(quadruplet.object)
+			if err != nil {
+				t.Fatalf("unable to parser/validate object yaml: %v\n%v", err, quadruplet.object)
+			}
+
+			// test RemoveItems
+			rmOut, err := pt.FromYAML(quadruplet.removeOutput)
+			if err != nil {
+				t.Fatalf("unable to parser/validate removeOutput yaml: %v\n%v", err, quadruplet.removeOutput)
+			}
+
+			rmGot := tv.RemoveItems(quadruplet.set)
+			if !value.Equals(rmGot.AsValue(), rmOut.AsValue()) {
+				t.Errorf("Expected\n%v\nbut got\n%v\n",
```

Comment:
Can we change the string here to differentiate betwee tne "Removed" and "Extracted"?

Reply:
done